### PR TITLE
Proxy must be configured autom. based on conf

### PR DESCRIPTION
**Pull request by laurovalente**
As of now, proxy is not automatically configured based on configuration. This change enhances it.

### CODE REVIEW

**Comment by laurovalente** [comment]
Another thing that could exist is the possibility to change the CURL configs, as it might be interesting:
Adding this line to the constructor would do the trick.

if( array_key_exists('curl', $configs))
            $this->curlOptions = $configs['curl'] + $this->curlOptions;

**Comment by jaypatel512** [comment]
Hey @laurovalente !

Thanks for the PR. I will go ahead and do a quick overview soon, and get back to you with the coments. 👍

**Comment by modGTB** [comment]
New function to set expiryBufferTime for OAuthTokenCredential to set the buffer time independent to session lifetime.

Therefore a new function in class OAuthTokenCredential like this:

`
    /**
     * Helper method to set expiryBufferTime
     *
     * @param $time
     */
    public function setExpiryBufferTime($time)
    {
        self::$expiryBufferTime = (int)$time;
    }

`

File: lib/PayPal/Core/PayPalHttpConfig.php [hunk 1]
```diff
@@ -68,6 +68,11 @@ public function __construct($url = null, $method = self::HTTP_POST, $configs = a
             //Remove the Cipher List for NSS
             $this->removeCurlOption(CURLOPT_SSL_CIPHER_LIST);
         }
+        if( array_key_exists('http.UseProxy',$configs) && $configs['http.UseProxy'] ){
+			if( !array_key_exists('http.Proxy',$configs) )
```
Comment (jaypatel512) [inline lib/PayPal/Core/PayPalHttpConfig.php]: Can we just make it such that if the `http.Proxy` exists, it means, we want to use proxy. Having additional `http.UseProxy` seems meaningless. Thoughts ?

File: lib/PayPal/Core/PayPalHttpConfig.php [hunk 2]
```diff
@@ -229,9 +234,10 @@ public function setHttpTimeout($timeout)
      * Set HTTP proxy information
      *
      * @param string $proxy
+     * @param array $otherConfigs you can set other proxy configurations
      * @throws PayPalConfigurationException
      */
-    public function setHttpProxy($proxy)
+    public function setHttpProxy($proxy,$otherConfigs=[CURLOPT_PROXYAUTH=>CURLAUTH_NTLM,CURLOPT_HTTPPROXYTUNNEL=>1,CURLOPT_SSL_VERIFYPEER=>0])
```
Comment (jaypatel512) [inline lib/PayPal/Core/PayPalHttpConfig.php]: We can move these $otherConfigs to something similar to [$defaultCurlOptions](https://github.com/laurovalente/PayPal-PHP-SDK/blob/master/lib/PayPal/Core/PayPalHttpConfig.php#L21)
such as `$defaultProxyCurlOptions`. This should allow developers to update them from the outside.

File: lib/PayPal/Core/PayPalHttpConfig.php (hunk 1 above)
Comment (laurovalente) [inline lib/PayPal/Core/PayPalHttpConfig.php]: Makes sense! :+1:

File: lib/PayPal/Core/PayPalHttpConfig.php (hunk 2 above)
Comment (laurovalente) [inline lib/PayPal/Core/PayPalHttpConfig.php]: Yes, that makes sense as well. Just did not think about this :+1:

**Comment by jaypatel512** [comment]
Hey @laurovalente !

Just a reminder ping, on if you could help us with the review changes. We can go ahead and merge the PR after that.

**Comment by jaypatel512** [comment]
Closing this PR based on inactivity. Please re-open if anyone interested helping us fix.

**Pull request closed without merging**